pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -19,110 +19,63 @@
 
 import cockpit from "cockpit";
 import React from "react";
+import { FormGroup, HelpBlock, TypeAheadSelect } from "patternfly-react";
 import PropTypes from "prop-types";
-import "./cockpit-components-file-autocomplete.css";
 import { debounce } from 'throttle-debounce';
 
 const _ = cockpit.gettext;
 
 export class FileAutoComplete extends React.Component {
     constructor(props) {
         super(props);
-        const value = props.value || "";
-        this.updateFiles(value);
+        this.updateFiles(props.value || "/");
         this.state = {
-            value: value,
-            directory: '',
-            directoryFiles: null,
+            directory: '', // The current directory we list files/dirs from
             displayFiles: [],
-            open: false,
             error: null,
+            hasFocus: false,
+            value: '',
         };
         this.allowFilesUpdate = true;
-        this.onChange = this.onChange.bind(this);
-        this.onMouseDown = this.onMouseDown.bind(this);
-        this.onChangeCallback = this.onChangeCallback.bind(this);
-        this.onBlur = this.onBlur.bind(this);
         this.updateFiles = this.updateFiles.bind(this);
-        this.updateIfDirectoryChanged = this.updateIfDirectoryChanged.bind(this);
         this.finishUpdate = this.finishUpdate.bind(this);
-        this.filterFiles = this.filterFiles.bind(this);
-        this.showAllOptions = this.showAllOptions.bind(this);
-        this.selectItem = this.selectItem.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.onInputChanged = this.onInputChanged.bind(this);
+
         this.debouncedChange = debounce(300, (value) => {
-            let cb = () => {
-                this.pendingCallback = false;
-                let stateUpdate = this.filterFiles(value);
-                this.setState(stateUpdate, () => this.onChangeCallback(value, { error: stateUpdate.error }));
+            let cb = (dirPath) => {
+                this.updateFiles(dirPath == '' ? '/' : dirPath);
             };
```

Comment:
nitpick: please let's not use these unnecessary { ; }. I. e. `let cb = dirPath => this.updateFiles(...);